app/controllers/rentals_controller.rb

```diff
@@ -0,0 +1,93 @@
+class RentalsController < ApplicationController
+
+  def checkout
+    rental = Rental.new(rental_params)
+
+    if rental.remove_movie
+      # if sufficient inventory to remove the movie, set checkout date and save
+      rental.checkout_date = rental.today
+      rental.save
+      # if valid save, return ok status
+      if rental.valid?
+        render(
+          json: rental, status: :ok
+        )
+      else
+        render(
+          json: { errors: rental.errors.messages },
+          status: :bad_request
+        )
+      end
+    else
+      render(
+        json: { ok: false, message: "Insufficient inventory of selected movie -- cannot checkout"},
+        status: :bad_request
+      )
+    end
+
+  end
+
+  def checkin
+    rental = Rental.where(["customer_id = ? and movie_id = ?", params[:customer_id], params[:movie_id]])
+
+    if rental.count == 0
+      render(
+        json: { errors: "No rental found" },
+        status: :not_found
+      )
+      return
+    elsif rental.count > 1
```

Comment:
This block looks like it could either be a model method or using an ennumerable.